### PR TITLE
Changes for seL4 loader policy

### PR DESCRIPTION
**Pull request by amstrnad**

### CODE REVIEW

File: osv/loader.dpl [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+module osv.loader:
+
+/* Load tags when loading program
```
Comment (natestuder) [inline osv/loader.dpl:29]: Nice commentary!

File: osv/loader.dpl [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+
+metadata:
+      Loader,
+      TagArray
+
+policy:
+  global loaderPol =
+        storeMemGrp(code == [+Loader], env == _, addr == _, val == _, mem == {} -> env = env, mem = val)
+      ^ storeMemGrp(code == [+Loader], env == _, addr == _, val == _, mem == _ -> env = env, mem = mem \/ val)
```
Comment (natestuder) [inline osv/loader.dpl:91]: Why is the union being stored?  I guess that might make the loader slightly more efficient, but results in the "uncleared tags" issue mentioned previously.

File: osv/loader.dpl [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      ^ loadMemGrp (code == [+Loader], env == _, addr == _, mem == _ -> env = env, res = mem)
+      ^ composeGrp (code == [+Loader], env == _, op1 == {},  op2== _ -> env = env, res = op2)
+      ^ composeGrp (code == [+Loader], env == _, op1 == _,  op2== {} -> env = env, res = op1)
```
Comment (natestuder) [inline osv/loader.dpl:94]: It looks like the old loader policy was slightly wrong here in how it handled op2.  Good catch.

File: osv/loader.dpl [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      ^ composeGrp (code == [+Loader], env == _, op1 == _,  op2== _ -> env = env, res = op1 \/ op2)
+      ^ mvGrp (code == [+Loader], env == _, op1 == _ -> env = env, res = op1 )
```
Comment (natestuder) [inline osv/loader.dpl:96]: What's the mvGrp needed for?

Comment (amstrnad) [reply osv/loader.dpl:96]: I've seen the generated assembly load memory into a register and then immediately moves it to another register.

File: osv/loader.dpl (hunk 2 above)
Comment (amstrnad) [reply osv/loader.dpl:91]: It saw it as a choice between leaving possibly extraneous tags or removing possibly relevant tags. I chose the simpler option to implement.

I think either would have worked in this case.

Comment (natestuder) [reply osv/loader.dpl:91]: If @ccasin is cool with it, I am.

File: osv/loader.dpl (hunk 4 above)
Comment (natestuder) [reply osv/loader.dpl:96]: Got it.

File: osv/riscv.dpl [hunk 5]
```diff
@@ -534,6 +536,19 @@ group:
         fmv.w.x
         fmv.d.x
         fmv.q.x
+    // atomicGrp
+        lr.w
+        sc.w
```
Comment (natestuder) [inline osv/riscv.dpl:541]: What about the `.d` variant of these opcodes?  Does seL4 really only use the `.w` variant?

Comment (amstrnad) [reply osv/riscv.dpl:541]: Apparently so - as noted in the policy-engine pr, we don't decode those variants yet either.

**Pull request merged**